src/go/k8s/apis/redpanda/v1alpha1/cluster_webhook.go

```diff
@@ -582,13 +588,15 @@ func (r *Cluster) ValidateDelete() error {
 	return nil
 }
 
+// TODO move this to networking package
 func (r *Cluster) checkCollidingPorts() field.ErrorList {
 	var allErrs field.ErrorList
 
 	type listenersPorts struct {
 		name                 string
 		port                 int
 		externalConnectivity bool
+		externalPort         *int
```

Comment:
nit: Why not treating the external ports as another listenerPort? I think that could simplify the method

Reply:
it would make it harder to distinguish between where external port collide and thus construct the message. But I like your idea. I would actually like to refactor this further and move it to the ports module so I can try to clean it up when I do that if it's fine with you. I'll work on that tomorrow and do it in another PR.

Reply:
Sounds good.